aten/src/ATen/cpu/vec256/vec256_qint.h

```diff
@@ -362,6 +362,26 @@ Vec256<c10::qint32> inline operator*(
 #endif
 }
 
+template <>
+Vec256<c10::qint32> inline operator+(
+    const Vec256<c10::qint32>& a,
+    const Vec256<c10::qint32>& b) {
+#ifdef __AVX2__
+  return _mm256_add_epi32(a, b);
+#else
+  // Pray the compiler can autovectorize this
+  int32_t a_vals[a.size()];
+  int32_t b_vals[b.size()];
+  a.store(a_vals);
+  b.store(b_vals);
+  int32_t result_vals[a.size()];
+  for (size_t i = 0; i < a.size(); ++i) {
+    result_vals[i] = a_vals[i] + b_vals[i];
+  }
+  return Vec256<c10::qint32>::loadu(result_vals);
+#endif
+}
+
```

Comment:
Can these be implemented as members on Vec256<c10:;qint32> instead of free functions?

Reply:
They are implemented the same way for float. I am not sure if there is a reason for them for them to be free functions. 
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cpu/vec256/vec256_float.h#L244-L262